docs/root/intro/arch_overview/security/ssl.rst

```diff
@@ -114,19 +114,37 @@ Certificate selection
 ---------------------
 
 :ref:`DownstreamTlsContexts <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.DownstreamTlsContext>` support multiple TLS
-certificates. These may be a mix of RSA and P-256 ECDSA certificates. The following rules apply:
+certificates. These may be a mix of RSA and P-256 ECDSA certificates for multiple services. The following rules apply:
+
+Certificate config/loading rules:
 
 * Only one certificate of a particular type (RSA or ECDSA) may be specified.
 * Non-P-256 server ECDSA certificates are rejected.
-* If the client supports P-256 ECDSA, a P-256 ECDSA certificate will be selected if one is present in the
-  :ref:`DownstreamTlsContext <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.DownstreamTlsContext>`
-  and it is in compliance with the OCSP policy.
-* If the client only supports RSA certificates, a RSA certificate will be selected if present in the
-  :ref:`DownstreamTlsContext <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.DownstreamTlsContext>`.
-* Otherwise, the first certificate listed is used. This will result in a failed handshake if the
-  client only supports RSA certificates and the server only has ECDSA certificates.
 * Static and SDS certificates may not be mixed in a given :ref:`DownstreamTlsContext
   <envoy_v3_api_msg_extensions.transport_sockets.tls.v3.DownstreamTlsContext>`.
+
+SNI matching rules:
+
+* If the client support SNI, a certificate with proper DNS SANs or Subject Common Name should be selected
+* If no certificate is matched to SNI, the connection is refused.
```

Comment:
Got it.

Reply:
@ggreenway Since we introduce SNI-based selection, we can't restrict there are 1 or 2 certs in total. We should restrict 1 or 2 certs(RSA, ESDSA) for each Server Name Pattern (DNS SAN entry or Subject Common Name). See code and doc for details.